miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleRestController.java

```diff
@@ -263,28 +269,65 @@ public SampleDto updateSample(@PathVariable("id") Long id, @RequestBody SampleDt
 
   /**
    * 
-   * @param identitiesSearches
+   * @param json
    *          String externalNames or Identity aliases
    * @param response
    * @return
    * @throws IOException
    */
   @RequestMapping(value = "/identities", method = RequestMethod.POST, headers = { "Content-type=application/json" })
   @ResponseBody
-  public Set<SampleDto> getIdentitiesBySearch(@RequestBody com.fasterxml.jackson.databind.JsonNode json,
+  public Set<SampleDto> getIdentityBySearch(@RequestBody com.fasterxml.jackson.databind.JsonNode json,
       HttpServletResponse response) throws IOException {
-    if (json.get("identitiesSearches").asText().length() == 0) {
-      throw new RestException("Must give search terms to look up identities", Status.BAD_REQUEST);
+    final JsonNode searchTerms = json.get("identitiesSearches");
+    final String project = (json.get("project") == null ? "" : json.get("project").asText());
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "project" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)